File: MyInfArith.py
import os 

class MyInfArith :
    def __init__(self, mode):
        self.file_path = ""
        self.mode = mode

    def compile(self) :
        os.system(f"javac -cp .:{self.file_path}target/aarithmetic.jar {self.file_path}myinfarith/MyInfArith.java")
    def run(self, op, x, y) :
        os.system(f"java -cp .:{self.file_path}target/aarithmetic.jar {self.file_path}myinfarith.MyInfArith {self.mode} {op} {x} {y}")   


def main() :

    mode = input("Enter the mode of operation: <int/float> \n")
    s = MyInfArith(mode)
    s.compile()
    print("Enter the operation to be performed: ")
    print("1. for ADDITION")
    print("2. for SUBTRACTION")
    print("3. for MULTIPLICATION")
    print("4. for DIVISION")
    op = int(input())
    if op == 1 :
        op = "add"
    elif op == 2 :
        op = "sub"
    elif op == 3 :
        op = "mul"
    elif op == 4 :
        op = "div"
    else :
        print("Invalid operation")
        return
    x = input("Enter the first number: ")
    y = input("Enter the second number: ")
    print("\nResult: ")
    s.run(op, x, y)

if __name__ == "__main__":
    main()

File: myinfarith/MyInfArith.java
package myinfarith;

import arbitraryarithmetic.*;

public class MyInfArith {
    public static void main( String[] args){
        if(args.length != 4){
            System.out.println("Usage: java MyInfArith <int/float> <add/sub/mul/div> <number1> <number2>");
            return;
        }
        String type = args[0];
        String operation = args[1];
        String num1 = args[2];
        String num2 = args[3];
        if(type.equals("int")){
            AInteger a1 = new AInteger(num1);
            AInteger a2 = new AInteger(num2);
            switch(operation){
                case "add":
                    System.out.println(a1.add(a1, a2).value);
                    break;
                case "sub":         
                    System.out.println(a1.sub(a1, a2).value);
                    break;
                case "mul":             
                    System.out.println(a1.mul(a1, a2).value);
                    break;
                case "div":
                    System.out.println(a1.div(a1, a2).value);
                    break;
                default:
                    System.out.println("Invalid operation. Use add, sub, mul, or div.");
                    break;
                }
        }
        else if(type.equals("float")){
            AFloat a1 = new AFloat(num1);
            AFloat a2 = new AFloat(num2);
            switch(operation){
                case "add":
                    System.out.println(a1.add(a1, a2).value);
                    break;
                case "sub":         
                    System.out.println(a1.sub(a1, a2).value);
                    break;
                case "mul":             
                    System.out.println(a1.mul(a1, a2).value);
                    break;
                case "div":
                    System.out.println(a1.div(a1, a2).value);
                    break;
                default:
                    System.out.println("Invalid operation. Use add, sub, mul, or div.");
                    break;
                }
        }
        else{
            System.out.println("Invalid type. Use AInteger or AFloat.");
        }

    }
    //end of project
}
